config/PropertyNames.xml

```diff
@@ -195,363 +195,364 @@ generated from the section label.
+        <property name="Lookup" languages="cpp,csharp,java"/>
+        <property name="Timeout" languages="cpp,csharp,java" default="300" />
+        <property name="RetryCount" languages="cpp,csharp,java" default="3" />
+        <property name="RetryDelay" languages="cpp,csharp,java" default="2000" />
+        <property name="Address" languages="cpp,csharp,java"/>
+        <property name="Port" languages="cpp,csharp,java" default="4061" />
+        <property name="Interface" languages="cpp,csharp,java" />
+        <property name="InstanceName" languages="cpp,csharp,java" default="IceLocatorDiscovery" />
+        <property name="Trace.Lookup" languages="cpp,csharp,java" default="0" />
     </section>
 
     <section name="IceBox">
-        <property name="InheritProperties" />
-        <property name="LoadOrder" />
-        <property name="PrintServicesReady" />
-        <property name="Service.[any]" />
-        <property name="Trace.ServiceObserver" />
-        <property name="UseSharedCommunicator.[any]" />
+        <property name="InheritProperties" languages="cpp,csharp,java" />
+        <property name="LoadOrder" languages="cpp,csharp,java" />
+        <property name="PrintServicesReady" languages="cpp,csharp,java" />
+        <property name="Service.[any]" languages="cpp,csharp,java" />
+        <property name="Trace.ServiceObserver" languages="cpp,csharp,java" />
+        <property name="UseSharedCommunicator.[any]" languages="cpp,csharp,java" />
     </section>
 
     <section name="IceBoxAdmin">
-        <property name="ServiceManager.Proxy" class="proxy"/>
+        <property name="ServiceManager.Proxy" class="proxy" languages="cpp" />
     </section>
 
     <section name="IceBridge">
-        <property name="Source" class="objectadapter" />
-        <property name="Target.Endpoints" />
-        <property name="InstanceName" default="IceBridge" />
+        <property name="Source" class="objectadapter" languages="cpp" />
+        <property name="Target.Endpoints" languages="cpp" />
+        <property name="InstanceName" default="IceBridge" languages="cpp" />
     </section>
 
     <section name="IceGridAdmin">
-        <property name="AuthenticateUsingSSL" />
-        <property name="MetricsConfig" />
-        <property name="Username" />
-        <property name="Password" />
-        <property name="Replica" />
-        <property name="Host" />
-        <property name="Port" />
-        <property name="InstanceName" default="IceGridAdmin" />
-        <property name="Server" class="objectadapter" />
-        <property name="Discovery.Address" />
-        <property name="Discovery.Interface" />
-        <property name="Discovery.Lookup" />
-        <property name="Discovery.Reply" class="objectadapter" />
-        <property name="Discovery.Locator" class="objectadapter" />
-        <property name="Trace.Observers" />
-        <property name="Trace.SaveToRegistry" />
+        <property name="AuthenticateUsingSSL" languages="cpp,java" />
```

Comment:
I suspect some properties are cpp-only or java-only, e.g. Trace.SaveToRegistry is java-only.